codegen/config/binder.go

```diff
@@ -167,6 +167,7 @@ type TypeReference struct {
 	Definition  *ast.Definition
 	GQL         *ast.Type
 	GO          types.Type
+	Target      types.Type
```

Comment:
Could we leave a comment that tries to explain the difference between Go and Type?

The target is the type of the physical field in the struct, and GO is the derived type when you combine it with the pointeryness required by the ast?

Reply:
Done - it's not even that, it's the difference between how to bind to the type, and how to traverse to the type from a field on another type, where sometimes that is due to gql AST requirements.

I wonder if there's an abstraction that represents both pointer juggling _and_ casting sorta like ` “x is satisfiable by y if i do a, b and c”` 🤔